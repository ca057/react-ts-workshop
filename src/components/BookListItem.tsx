import React from "react";

import { Book } from "../domain/types";

interface BookListItemProps {
  book: Book;
  onClick: (book: Book) => void;
}

const BookListItem: React.FC<BookListItemProps> = ({ book, onClick }) => (
  <button onClick={() => onClick(book)}>{book.title}</button>
);

export default BookListItem;
