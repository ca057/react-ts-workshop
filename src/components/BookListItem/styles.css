.booklist__item {
    padding: 5px;
    background-color:mintcream;
}
