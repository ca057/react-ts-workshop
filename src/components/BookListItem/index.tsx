import React from "react";
import { Link } from "react-router-dom";

import { Book } from "../../domain/types";
import "./styles.css";

interface BookListItemProps {
  book: Book;
}

const BookListItem: React.VFC<BookListItemProps> = ({ book }) => (
  <Link to={`/books/${book.isbn}`}>
    <div className="booklist__item">{book.title}</div>
  </Link>
);

export default BookListItem;
