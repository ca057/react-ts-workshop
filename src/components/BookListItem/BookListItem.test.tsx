import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import BookListItem from ".";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

describe("components/BookListItem", () => {
  const book = { title: "My first book", price: 11.11 };

  test(`renders a booklist item correctly`, () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <BookListItem book={book} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders book with title", () => {
    render(<BookListItem book={book} />, {
      wrapper: BrowserRouter,
    });

    expect(screen.getByText(book.title)).toBeTruthy();
  });
});
