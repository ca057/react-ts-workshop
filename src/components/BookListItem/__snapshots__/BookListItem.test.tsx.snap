// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BookListItem renders a booklist item correctly 1`] = `
<a
  href="/books/undefined"
  onClick={[Function]}
>
  <div
    className="booklist__item"
  >
    My first book
  </div>
</a>
`;
